src/main/java/org/broadinstitute/dsm/route/PatchRoute.java

```diff
@@ -537,4 +547,19 @@ private NameValue setAdditionalValue(String additionalValue, @NonNull Patch patc
             throw new RuntimeException("DBElement not found in ColumnNameMap: " + additionalValue);
         }
     }
+
+    private void triggerParticipantEvent(DDPInstance ddpInstance, Patch patch, Value action){
+        inTransaction((conn) -> {
+            Optional<EventTypeDto> eventType = EventTypeDao.getEventTypeByEventTypeAndInstanceId(conn, action.getName(), ddpInstance.getDdpInstanceId());
```

Comment:
having `inTransaction` in `getEventTypeByEventTypeAndInstanceId` instead of passing connection will make business logic classes to be more loosely coupled.